### PR TITLE
Update CHANGELOG.md to include Helm image change

### PR DESCRIPTION
**Pull request by hugoShaka**
We had 2 support questions regarding the chart image changes in v13 and I was not able to find an item in the changelog explaining the changes.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -172,6 +172,23 @@ Teleport container images are no longer being published to it.
...
 Users should use the public ECR registry: https://goteleport.com/docs/installation/#docker.
 
+#### Helm chart uses `distroless`-based container image by default
+
+Starting with Teleport 13, the Helm charts `teleport-cluster` and `teleport-kube-agent`
+are deploying distroless Teleport images by default. Those images are slimmer
+and more secure but contain less tooling (e.g. neither `bash` nor
+`apt` are available).
+
+The debian-based images are deprecated and will be removed in Teleport 14.
```
Comment (ptgott) [inline CHANGELOG.md:182]:
```suggestion
The Debian-based images are deprecated and will be removed in Teleport 14.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -172,6 +172,23 @@ Teleport container images are no longer being published to it.
...
+The chart image can be reverted back to the debian-based images by setting:
```
Comment (ptgott) [inline CHANGELOG.md:183]:
```suggestion
The chart image can be reverted back to the Debian-based images by setting:
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -172,6 +172,23 @@ Teleport container images are no longer being published to it.
...
+```
```
Comment (ptgott) [inline CHANGELOG.md:184]:
```suggestion
```yaml
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -172,6 +172,23 @@ Teleport container images are no longer being published to it.
...
+image: "public.ecr.aws/gravitational/teleport"
+```
+
+For debugging purposes, a "debug" image is available and contains busybox,
```
Comment (ptgott) [inline CHANGELOG.md:188]:
```suggestion
For debugging purposes, a "debug" image is available and contains BusyBox,
```

**Review by ptgott — APPROVED** [review]
Approved with minor suggestions

**Pull request merged**